gpu/src/metal/fft/polynomial.rs

```diff
@@ -7,63 +7,61 @@ use lambdaworks_math::{
     polynomial::Polynomial,
 };
 
-use super::{errors::FFTMetalError, helpers::log2, ops::*};
-
-pub trait MetalFFTPoly<F: IsTwoAdicField> {
-    fn evaluate_fft_metal(&self) -> Result<Vec<FieldElement<F>>, FFTMetalError>;
-    fn evaluate_offset_fft_metal(
-        &self,
-        offset: &FieldElement<F>,
-        blowup_factor: usize,
-    ) -> Result<Vec<FieldElement<F>>, FFTMetalError>;
-    fn interpolate_fft_metal(
-        fft_evals: &[FieldElement<F>],
-    ) -> Result<Polynomial<FieldElement<F>>, FFTMetalError>;
-}
-
-impl<F: IsTwoAdicField> MetalFFTPoly<F> for Polynomial<FieldElement<F>> {
-    /// Evaluates this polynomial using parallel FFT (so the function is evaluated using twiddle factors),
-    /// in Metal.
-    fn evaluate_fft_metal(&self) -> Result<Vec<FieldElement<F>>, FFTMetalError> {
-        let metal_state = MetalState::new(None).unwrap();
-        let order = log2(self.coefficients().len())?;
-        let twiddles = gen_twiddles(order, RootsConfig::BitReverse, &metal_state)?;
+use super::{helpers::log2, ops::*};
 
-        fft(self.coefficients(), &twiddles, &metal_state)
-    }
-
-    /// Evaluates this polynomial using parallel FFT in an extended domain by `blowup_factor` with an `offset`, in Metal.
-    /// Usually used for Reed-Solomon encoding.
-    fn evaluate_offset_fft_metal(
-        &self,
-        offset: &FieldElement<F>,
-        blowup_factor: usize,
-    ) -> Result<Vec<FieldElement<F>>, FFTMetalError> {
-        let metal_state = MetalState::new(None).unwrap();
-        let scaled = self.scale(offset);
-
-        fft_with_blowup(scaled.coefficients(), blowup_factor, &metal_state)
-    }
+pub fn evaluate_fft_metal<F>(
+    poly: &Polynomial<FieldElement<F>>,
+) -> Result<Vec<FieldElement<F>>, MetalError>
+where
+    F: IsTwoAdicField,
+{
+    let metal_state = MetalState::new(None).unwrap();
+    let order = log2(poly.coefficients.len())?;
```

Comment:
Like [this line](https://github.com/lambdaclass/lambdaworks/pull/266/files#diff-fb8da2995a9269ce64ad6de87fcd126a4337e30a647852c57a917c2b1e470baaR36), Can this be replaced with `trailing_zeros`?

Reply:
Theres a validation done before https://github.com/lambdaclass/lambdaworks/blob/main/gpu/src/metal/fft/helpers.rs#L3